utils/ibdmp-decode

```diff
@@ -0,0 +1,261 @@
+#!/usr/bin/python3
+
+import base64
+import collections
+import hashlib
+import re
+import sys
+
+DEBUG = False
+RE_HEADER = r'_ibdmp:[0-9]+/[0-9]+[|]chunks=[0-9]+,md5=[0-9a-f]+[|]'
+RE_CHUNK = r'_ibdmp:[0-9]+/[0-9]+[|][0-9]+:[A-Za-z0-9+/=]+[|]'
+
+
+def USAGE():
+    self = sys.argv[0].split('/')[-1]
+    lines = [
+        "usage: %s path/to/console.log path/to/target.tar.xz" % self,
+        "",
+        "Decode debug tarball emited by leapp's initramfs in-band",
+        "console debugger, ibdmp().",
+    ]
+    sys.stderr.writelines('%s\n' % l for l in lines)
+    sys.exit(2)
+
+
+def LOG_DEBUG(msg):
+    if DEBUG:
+        sys.stderr.write('DEBUG: %s\n' % msg)
+
+
+def LOG_WARN(msg):
+    sys.stderr.write('%s\n' % msg)
+
+
+class IbdmbDecodeError(ValueError):
+    pass
+
+
+class UsageError(ValueError):
+    pass
+
+
+class _Chunk:
+
+    @classmethod
+    def from_raw1(cls, raw_chunk):
+        assert raw_chunk.startswith('_ibdmp:')
+        areas = raw_chunk.split('|')
+        parts = areas[1].split(':')
+        return cls(
+            ordinal=int(parts[0]),
+            payload=str(parts[1]),
+        )
+
+    def __init__(self, ordinal, payload):
+        self.ordinal = ordinal
+        self.payload = payload
+
+    def __hash__(self):
+        return hash((self.ordinal, self.payload))
```

Comment:
Why is the `ordinal` part of the hash? Isn't the `payload` the identity element (content-addressable)?

Reply:
Not if it's corrupted.

(That is, if i remember my thinking -- from 5 months ago -- correctly.)

Reply:
Umm, you are probably right.

Reply:
Hmm, is this part even needed? I don't see `_Chunk` being hashed anywhere.

Reply:
I think the hashability is needed because the use of `collections.Counter` relies on it.

Ie. check out `ChunkCounter.__init__` -- I'm loading the chunks into a huge dictionary (`ChunkCounter._bagset`) where keys are chunk ordinals, and each value is `collections.Counter` -- that way I can load chunks in any order and `collections.Counter` will do the maths.

You're probably right that the `ordinal` does not need to be part of the hash just for this to work.  OTOH, I would not say the payload is content-addressable: one could have "same" chunk at different positions and semantically it would not really be "the same chunk" -- so even if it's not functionally necessary (`_Chunk` is internal, after all) it seems more correct to include it.

WDYY?

Reply:
So on line 173 I see:
```
self._bagset[c.ordinal].update([c.payload])
```

So it seems that `c` doesn't need to be hashable because it's members are put into the structure and not `c` itself?

Reply:
nvm it probably gets hashed at line 130

Reply:
L130 is hashing `Header.chunks` but that's just int

Reply:
IOW, you're right, neither `Header` nor `_Chunk` actually need `__hash__` -- it's likely just remainder from some earlier attempt ising `set` or something.